apps/XMRig/install

```diff
@@ -0,0 +1,287 @@
+    if [ "$tls" == false ];then
+      local tls_options='Disabled!Supported by mining pool'
+    else
+      local tls_options='Supported by mining pool!Disabled'
+    fi
+    
+    if [ "$keepalive" == false ];then
+      local keepalive_options='Disabled!Supported by mining pool'
+    else
+      local keepalive_options='Supported by mining pool!Disabled'
+    fi
+    
+    #One window to choose everything
+    local output
+    output="$(yad --center --no-escape --form --separator='\n' \
+      --title="Create miner config" \
+      --window-icon="$(dirname "$0")/icon-64.png" \
+      --button="Skip this setup":1 \
+      --button=OK:0 \
+      --field="Name your miner" "$miner_name" \
+      --field="Wallet address" "$wallet_address" \
+      --field="Mining pool" "$mining_pool" \
+      --field="SSL/TLS:CB" "$tls_options" \
+      --field="Keepalive packets:CB" "$keepalive_options")"
+    local button=$?
+    
+    if [ $button == 0 ];then
+      miner_name="$(sed -n 1p <<<"$output")"
+      wallet_address="$(sed -n 2p <<<"$output")"
+      mining_pool="$(sed -n 3p <<<"$output")"
+      tls="$(sed -n 4p <<<"$output" | sed 's/Disabled/false/' | sed 's/Supported by mining pool/true/')"
+      keepalive="$(sed -n 5p <<<"$output" | sed 's/Disabled/false/' | sed 's/Supported by mining pool/true/')"
+      
+      if [ -z "$wallet_address" ];then
+        warning "Wallet address cannot be left blank!"
+      elif [ -z "$mining_pool" ];then
+        warning "Mining pool cannot be left blank!"
+      else
+        local save_settings='yes'
+        break
+      fi
+    else
+      local save_settings='no'
+      break
+    fi
+  done
+  
+  if [ "$save_settings" == 'yes' ];then
+    # Save choices to xmrig.json
+    status "Saving to $HOME/.config/xmrig.json"
+    echo "Contents of file:"
+    cat <<EOT | tee $HOME/.config/xmrig.json 1>&2 || error 'Failed to create $HOME/.config/xmrig.json!'
+{
+    "autosave": true,
+    "background": null,
+    "cpu": true,
+    "opencl": false,
+    "cuda": false,
+    "rx": [-1, -1],
+    "pools": [
+        {
+            "url": "${mining_pool}",
+            "user": "${wallet_address}",
+            "rig-id": "${miner_name}",
+            "keepalive": ${keepalive},
+            "tls": ${tls}
+        }
+    ]
+}
+EOT
+  else #User skipped setup wizard
+    echo -e "You chose to not set up a configuration for xmrig."
+  fi
+  
+} # End setup_wizard()
+
+cd
+rm -rf $HOME/xmrig
+
+#Compile xmrig; set to false to skip compilation
+if true;then
+  install_packages git build-essential cmake libuv1-dev libssl-dev libhwloc-dev yad xprintidle || exit 1
+  
+  git_clone https://github.com/MoneroOcean/xmrig || error 'Failed to clone repository!'
+  
+  mkdir $HOME/xmrig/build; cd $HOME/xmrig/build || error 'Failed to create and cd into build directory!'
+  
+  status "Preparing to compile..."
+  
+  #Optimize for specific CPU architecture, if detected. Get full list with 'gcc --target-help -mtune=foo' 
```

Comment:
on arm (unlike on x86), using `-mcpu=native` will automatically enable all proper -march and -mtune

arm did a full writeup but the TLDR is use `-mcpu=native`
https://community.arm.com/arm-community-blogs/b/tools-software-ides-blog/posts/compiler-flags-across-architectures-march-mtune-and-mcpu

Reply:
Done.